src/scripts/selectors.js

```diff
@@ -1,10 +1,67 @@
 import {
+    filter,
     find,
     split
 } from 'lodash'
+import accounting from 'accounting-js'
+import moment from 'moment'
 
 import { CURRENCIES } from '../constants'
 
+export const formatAmount = (props) => {
+    const {
+        amount,
+        currencyInformation
+    } = props
+    return accounting.formatMoney(
+        amount,
+        {
+            symbol: currencyInformation['symbol'],
+            thousand: currencyInformation['thousand'],
+            decimal: currencyInformation['decimal'],
+            format: '%s %v'
+        }
+    )
+}
+
+export const getAllocatedContributors = ({ allocations }) => {
+    const contributorsAllocated = []
+    allocations.map(a => {
+        if (!contributorsAllocated.includes(a.contributor)) {
+            contributorsAllocated.push(a.contributor)
+        }
+    })
+    return contributorsAllocated
+}
+
+export const getActiveAndUpcomingAllocations = ({ activeOnly, allocations, upcomingOnly }) => {
+    const desiredAllocations = []
+    allocations.map(a => {
+        if (activeOnly) {
+            if (moment(a['start_date'], 'x').isBefore(moment()) && moment(a['end_date'], 'x').isAfter(moment())) {
+                desiredAllocations.push(a)
+            }
+        } else if (upcomingOnly) {
+            if (moment(a['start_date'], 'x').isAfter(moment())) {
+                desiredAllocations.push(a)
+            }
+        } else if (moment(a['end_date'], 'x').isAfter(moment())) {
+            desiredAllocations.push(a)
+        }
+    })
+    return desiredAllocations
```

Comment:
Just FYI consider using `allocations.filter` for this if there are not transformations needed on the objects in the array.

(Or `reduce` if both filtering and re-mapping/transforming is needed)